### PR TITLE
Header Update closes #30

### LINKED ISSUE #30: Header for homepage
Create the header for homepage

### PR DESCRIPTION
**Pull request by MiguelFieira**
## Changes

- Improved the header
- Added Dummy Content blocks

### How to Test
- Pull form branch `30-header-
  - Run command `npm ci` so you have the latest dependency 
  - Run command `npm run dev` to startup the Local Development Environment Server

### CODE REVIEW

File: src/components/button/button.scss [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    cursor: pointer;
+    background-color: $primary;
+    color: white;
+    text-align: center;
+}
+
+.inverted {
+    background-color: white;
+    color: $primary;
+}
```
Comment (MiguelFieira) [inline src/components/button/button.scss]: endline

File: src/components/index.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import Button from './button'
+import Hamburger from './hamburger'
+
+export {
+    Button,
+    Hamburger
+}
```
Comment (MiguelFieira) [inline src/components/index.js]: endline

File: src/layouts/header/header.scss [hunk 3]
```diff
@@ -1,55 +1,94 @@
...
     }
 }
+
+.hamburger {
+    margin-right: 20px;
+    display: none;
+    @include mobile-all {
+        display: block;
+    }
+}
```
Comment (MiguelFieira) [inline src/layouts/header/header.scss]: endline

File: src/layouts/index.js [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import Header from './header'
+import SideNavigation from './sideNavigation'
 import Main from './main'
 
 export {
     Header,
+    SideNavigation,
     Main
 }
```
Comment (MiguelFieira) [inline src/layouts/index.js:11]: endline

File: src/stores/defaultData.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  .model('DefaultDataStore', {
+    projectName: types.optional(types.string, 'Athena')
+  })
+  .actions(self => ({
+    reset () {
+      self.projectName = 'Athena'
+    }
+  }))
+
+export default DefaultDataStore
```
Comment (MiguelFieira) [inline src/stores/defaultData.js:13]: endline

File: src/stores/ui.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  .actions(self => ({
+    reset () {
+      self.subNavOpen = false
+    },
+    toggleSubNav () {
+      self.subNavOpen = !self.subNavOpen
+    }
+  }))
+
+export default UIStore
```
Comment (MiguelFieira) [inline src/stores/ui.js]: endline

File: src/styles/breakpoints.scss [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    @media all and (min-width: 768px) and (max-width: 1199px) {
+        @content;
+    }
+}
+
+@mixin mobile-all {
+    @media all and (max-width: 768px) {
+        @content;
+    }
+}
```
Comment (MiguelFieira) [inline src/styles/breakpoints.scss]: endline

File: src/styles/index.scss [hunk 8]
```diff
@@ -0,0 +1,5 @@
+// Only imports
+@import "./variables";
+@import "./primary";
+@import "./breakpoints";
+@import "./mixins";
```
Comment (MiguelFieira) [inline src/styles/index.scss]: endline

File: src/styles/mixins.scss [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+            user-select: none; /* Non-prefixed version, currently
+                                  supported by Chrome and Opera */
+  }
+
+@mixin transition-ease ($amount) {
+    -webkit-transition: all $amount ease-in-out;
+    -ms-transition: all $amount ease-in-out;
+    -o-transition: all $amount ease-in-out;
+    transition: all $amount ease-in-out;
+}
```
Comment (MiguelFieira) [inline src/styles/mixins.scss]: endline

File: tsconfig.json [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+    "compilerOptions": {
+        "experimentalDecorators": true,
+        "allowJs": true
+    }
+}
```
Comment (MiguelFieira) [inline tsconfig.json]: endline

**Review by MiguelFieira — COMMENTED** [review]
Minor Linting Errors

File: src/components/hamburger/hamburger.scss [hunk 11]
```diff
@@ -0,0 +1,33 @@
+@import '../../styles/index.scss';
+
+.hamburger {
+  cursor: pointer;
```
Comment (MiguelFieira) [inline src/components/hamburger/hamburger.scss:4]: Add whiteline

File: src/components/profileBar/profilebar.scss [hunk 12]
```diff
@@ -0,0 +1,60 @@
+@import '../../styles/index.scss';
+
+.profile {
+    @include noselect;
+    display: flex;
+    background-color: $darker;
+    height: 75px;
+    &Picture {
```
Comment (MiguelFieira) [inline src/components/profileBar/profilebar.scss:10]: Add whiteline

File: src/components/profileBar/profilebar.scss [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+        background-image: url('../../static/dummyprofile.jpg');
+        background-size: cover;
+        background-position: center;
+        width: 75px;
+        height: 75px;
+        position: relative;
+    }
```
Comment (MiguelFieira) [inline src/components/profileBar/profilebar.scss:18]: Add whiteline

File: src/components/profileBar/profilebar.scss [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+    &Info {
+        width: calc(100% - 75px);
+        padding: $header-spacing;
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        &UserName {
```
Comment (MiguelFieira) [inline src/components/profileBar/profilebar.scss:31]: Add whiteline

File: src/components/profileBar/profilebar.scss [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+            overflow: hidden;
+            white-space: nowrap;
+            text-overflow: ellipsis;
+            color: $background;
+            font-size: 1em;
+        }
+        &UserRole {
```
Comment (MiguelFieira) [inline src/components/profileBar/profilebar.scss:39]: Add whiteline

File: src/components/profileBar/profilebar.scss [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+            overflow: hidden;
+            white-space: nowrap;
+            text-overflow: ellipsis;
+            font-size: 0.75em;
+            color: $primary;
+        }
+    }
+    &Level {
```
Comment (MiguelFieira) [inline src/components/profileBar/profilebar.scss:48]: Add whiteline

File: src/components/profileBar/profilebar.scss [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+        position: absolute;
+        bottom: 0;
+        right: 0;
+        justify-content: center;
+        height: 25px;
+        min-width: 25px;
+        width: auto;
+        text-align: center;
+        background-color: $primary;
+        &Number {
```
Comment (MiguelFieira) [inline src/components/profileBar/profilebar.scss:63]: Add whiteline

File: src/components/profileBar/profilebar.scss [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+            font-family: $sidefont;
+            color: white;
+            font-size: 1em;
+            line-height: 0.8em;
+            margin-bottom: -2px;
+            padding: 0 6px;
+        }
+    }
+}
```
Comment (MiguelFieira) [inline src/components/profileBar/profilebar.scss]: Endline

File: src/layouts/header/header.scss [hunk 19]
```diff
@@ -1,55 +1,94 @@
...
+        max-width: $wrapper-width;
+        margin: 0 auto;
     }
-    @media screen and (min-width: 801px) {
-        font-size: $nr-logo-fontsize;
-        line-height: $nr-logo-fontsize * 0.75;
+}
+
+.filler {
+    // Added everything necessary to hide and
```
Comment (MiguelFieira) [inline src/layouts/header/header.scss:29]: Remove Comment

File: src/layouts/header/header.scss [hunk 20]
```diff
@@ -1,55 +1,94 @@
...
+    }
 }
 
 .menu {
-    &__ul {
+    @include mobile-sm {
+        display: none;
+    }
+    margin-left: auto;
+    &Ul {
```
Comment (MiguelFieira) [inline src/layouts/header/header.scss:89]: Add whiteline

File: src/layouts/header/header.scss [hunk 21]
```diff
@@ -1,55 +1,94 @@
...
         align-items: center;
         letter-spacing: 1px;
         font-weight: bold;
     }
-    &__link {
-        margin: 0 $nr-sm-fontsize;
-        cursor: pointer;
-    }
-    &__signup {
+    &Item {
```
Comment (MiguelFieira) [inline src/layouts/header/header.scss:100]: Add whiteline

File: src/layouts/sideNavigation/sideNavigation.scss [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+    width: 0px;
+    max-width: 0vw;
+    overflow-x: hidden;
+    background-color: $dark;
+    height: 100vh;
+    z-index: 2;
+    position: fixed;
+}
+
+.filler {
```
Comment (MiguelFieira) [inline src/layouts/sideNavigation/sideNavigation.scss]: We could combine this in a mixin sins some of the code is also used in `header.scss`

File: src/layouts/sideNavigation/sideNavigation.scss [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+        transition: 1s;
+        width: 325px;
+        max-width: 75vw;
+    }
+}
+
+.menu {
+    width: 325px;
+    max-width: 75vw;
+    &Ul {
```
Comment (MiguelFieira) [inline src/layouts/sideNavigation/sideNavigation.scss]: Add whiteline

File: src/layouts/sideNavigation/sideNavigation.scss [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+        list-style-type: none;
+        text-transform: uppercase;
+        font-size: 1em;
+    }
+    &Item {
```
Comment (MiguelFieira) [inline src/layouts/sideNavigation/sideNavigation.scss]: Add whiteline

File: src/layouts/sideNavigation/sideNavigation.scss [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+        font-weight: bold;
+        letter-spacing: 1px;
+        padding: $header-spacing;
+        transition: 0.2s;
+        cursor: pointer;
+    }
+    &Link {
```
Comment (MiguelFieira) [inline src/layouts/sideNavigation/sideNavigation.scss]: Add whiteline

**Review by MiguelFieira — COMMENTED** [review]
See Comments

**Comment by AjobK** [comment]
Oh no. I knew I would get 3 million linting comments from Miguel...

File: src/layouts/header/header.scss (hunk 19 above)
Comment (AjobK) [reply src/layouts/header/header.scss:29]: Added this to make sure everyone understands that piece of code

**Pull request closed without merging**